dot/parachain/candidate-validation/candidate_validation.go

```diff
@@ -43,9 +43,19 @@ func NewCandidateValidation(overseerChan chan<- any) *CandidateValidation {
 }
 
 // Run starts the CandidateValidation subsystem
-func (cv *CandidateValidation) Run(context.Context, chan any, chan any) {
-	cv.wg.Add(1)
-	go cv.processMessages(&cv.wg)
+func (cv *CandidateValidation) Run(context.Context, context.CancelFunc, chan any, chan any) {
+	for {
+		select {
+		case msg := <-cv.OverseerToSubsystem:
+			logger.Debugf("received message %v", msg)
+			err := cv.processMessages(msg)
+			if err != nil {
+				logger.Error(err.Error())
+			}
+		case <-cv.stopChan:
```

Comment:
I do not see any place where this stop channel is getting initiated or used (written to).

Reply:
I have just moved the code. I have not written any logic other than passing the `ctx` and `cancel` to the Run method.
But still, I will look into this and try to improve

Reply:
Removed `stopChan` and used `ctx.Done()` instead